File: RubyCivitas/lib/casilla_impuesto.rb
require_relative 'casilla'
require_relative 'jugador'

module Civitas
  class Casilla_impuesto < Casilla
    
    def initialize(nombre,importe)
      
      super(nombre)
      @importe = importe
      
    end
    
    def recibe_jugador_impuesto(actual, todos)

      if jugador_correcto(actual,todos)
        informe(actual, todos)
        todos[actual].paga_impuesto(@importe)
      end
    end
    
    def to_string
      
      "[Casilla] -> #{@nombre}\n[Tipo] -> Impuesto\n[Importe] -> #{@importe}\n"
      
    end
    
    attr_reader :importe
  end
end

File: civitasjava/src/Civitas/CasillaImpuesto.java
package civitas;

import java.util.ArrayList;


public class CasillaImpuesto extends Casilla{
    private float importe;
    
    CasillaImpuesto(float c, String n){
        super(n);
        importe = c;
    }
    
    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            todos.get(actual).pagaImpuesto(importe);
        }
    }
    
    @Override
    public String toString(){
        return "Casilla impuesto{ Nombre: "+this.getNombre() +"importe: " 
                +importe + "}";
    }
}
